File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/domain/community/posting/service/ReadPostingService.java
package io.dustin.apps.board.domain.community.posting.service;

import io.dustin.apps.board.domain.community.posting.model.Posting;
import io.dustin.apps.board.domain.community.posting.model.dto.PostingDto;
import io.dustin.apps.board.domain.community.posting.model.dto.PostingListDto;
import io.dustin.apps.board.domain.community.posting.repository.PostingRepository;
import io.dustin.apps.common.exception.DataNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class ReadPostingService {

    private final PostingRepository postingRepository;

    @Transactional(readOnly = true)
    public PostingDto getPosting(long userId, long postingId) {
        return postingRepository.getPosting(userId, postingId);
    }

    @Transactional(readOnly = true)
    public List<PostingListDto> getPostingList(long userId, Long nextId, int size) {
        return postingRepository.getPostingList(userId, nextId, size);
    }

    @Transactional(readOnly = true)
    public Posting findById(long postingId) {
        Optional<Posting> optional = this.postingRepository.findById(postingId);
        if(optional.isPresent()) {
            return optional.get();
        } else {
            return null;
        }
    }

    @Transactional(readOnly = true)
    public Posting findByIdOrThrow(long postingId) {
        Optional<Posting> optional = this.postingRepository.findById(postingId);
        if(optional.isPresent()) {
            return optional.get();
        } else {
            throw new DataNotFoundException("""
                    id [#1]로 조회된 게시물이 없습니다.""".replace("#1", String.valueOf(postingId)));
        }
    }



}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/community/posting/service/ReadPostingService.kt
package io.dustin.apps.board.domain.community.posting.service

import io.dustin.apps.board.domain.community.posting.model.Posting
import io.dustin.apps.board.domain.community.posting.repository.PostingRepository
import io.dustin.apps.common.exception.DataNotFoundException
import io.dustin.apps.common.repository.findByIdOrThrow
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ReadPostingService (
    private val postingRepository: PostingRepository
) {

    /**
     * db 캐싱에 사용되는 임시 메소드
     */
    fun TestFindAll() =
        postingRepository.findAll()

    /**
     * 게시물 상세보기
     */
    fun getPosting(userId: Long, postingId: Long) =
        postingRepository.getPosting(userId, postingId)

    /**
     * 게시물 리스트 가져오기
     */
    fun getPostingList(userId: Long, recordsCount: Long, nextId: Long?) =
        postingRepository.getPostingList(userId, recordsCount, nextId)


    fun findByIdOrNull(postingId: Long): Posting? {
        return postingRepository.findByIdOrNull(postingId)
    }

    fun findByIdOrThrow(postingId: Long): Posting {
        return postingRepository.findByIdOrThrow(postingId)
    }
}
